### PR TITLE
Add support for insulin, physicalActivity, food and reportedState (BACK-1810, BACK-1692)

### PR DESCRIPTION
**Pull request by gniezen**
Addresses [BACK-1810] and [BACK-1692].

[BACK-1810]: https://tidepool.atlassian.net/browse/BACK-1810

[BACK-1692]: https://tidepool.atlassian.net/browse/BACK-1692

### CODE REVIEW

File: lib/schema/physicalActivity.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+var schema = require('./schema.js');
+
+var idFields = ['type', 'deviceId', 'time'];
+schema.registerIdFields('physicalActivity', idFields);
+
+module.exports = schema.makeHandler('physicalActivity', {
+  schema: {
+    duration: schema.and(schema.isObject, schema.ensureSchemaFn('duration', {
+      value: schema.isNumber,
```
Comment (pazaan) [inline lib/schema/physicalActivity.js:28]: This doesn't quite [match the behaviour in platform](https://github.com/tidepool-org/platform/blob/25bfa5aaf9702b1d6a0eebe08798285c33868183/data/types/activity/physical/duration.go#L60-L72), which has more validation around the values depending on the units. If we wanted to replicate this behaviour, we may have to do a higher-level `schema.or`.

File: lib/schema/reportedState.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+'use strict';
+
+var schema = require('./schema.js');
+
+var idFields = ['type', 'deviceId', 'time'];
+schema.registerIdFields('reportedState', idFields);
+
+var stateArraySchema = schema.isArrayWithValueSchema({
+  state: schema.isString
```
Comment (pazaan) [inline lib/schema/reportedState.js:26]: In `platform`, there are a [few fixed states](https://github.com/tidepool-org/platform/blob/25bfa5aaf9702b1d6a0eebe08798285c33868183/data/types/state/reported/state.go#L14-L18) (ie, `state` must be one of the fixed states). If the state is anything other than one of the fixed states, then `state` should be "other", and an additional field called `stateOther` should exist, with a string of maximum length 100.
If you wanted to replicate this as simply as possible, you could do it like this:
```
var stateArraySchema = schema.isArrayWithValueSchema({
  state: schema.in('other'),
  stateOther: schema.isStringOfMaxLength(100)
});
```

**Review by pazaan — CHANGES_REQUESTED** [review]
Suggestions to match `platform` behaviour.

**Review by pazaan — DISMISSED** [review]
LGTM! :shipit:

**Comment by gniezen** [comment]
@tjotala Would you mind just re-approving this PR now that Lennart doesn't have write access anymore?

**Review by toddkazakov — DISMISSED** [review]

**Comment by gniezen** [comment]
Will need to deploy to QA1, as that's where the test accounts were made.

**Review by toddkazakov — APPROVED** [review]
LGTM

**Pull request merged**